code/modules/hivemind/mobs.dm

```diff
@@ -0,0 +1,913 @@
+	attacktext = "bangs with his head"
+	universal_speak = TRUE
+	speak_chance = 5
+	var/malfunction_chance = 5
+	var/ability_cooldown = 30 SECONDS
+	var/list/target_speak = list()			//this is like speak list, but when we see our target
+
+	//internals
+	var/obj/structure/hivemind_machine/master
+	var/special_ability_cooldown = 0		//use ability_cooldown, don't touch this
+
+
+	New()
+		..()
+		//here we change name, so design them according to this
+		name = pick("strange ", "unusual ", "odd ", "undiscovered ", "an interesting ") + name
+
+//It's sets manually
+/mob/living/simple_animal/hostile/hivemind/proc/special_ability()
+	return
+
+
+/mob/living/simple_animal/hostile/hivemind/proc/is_on_cooldown()
+	if(world.time >= special_ability_cooldown)
+		return FALSE
+	return TRUE
+
+
+//simple shaking animation, this one move our target horizontally
+/mob/living/simple_animal/hostile/hivemind/proc/shake_the(atom/target)
+	var/init_px = target.pixel_x
+	animate(target, pixel_x=init_px + 10*pick(-1, 1), time=1)
+	animate(pixel_x=init_px, time=8, easing=BOUNCE_EASING)
+
+
+//That's just stuns us for a while and start second proc
+/mob/living/simple_animal/hostile/hivemind/proc/mulfunction()
+	stance = HOSTILE_STANCE_IDLE //it give us some kind of stun effect
+	target_mob = null
+	walk(src, FALSE)
+	var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread()
+	sparks.set_up(3, 3, loc)
+	sparks.start()
+	shake_the(src)
+	if(prob(30))
+		say(pick("Fu-ue-ewe-eweu-u-uck!", "A-a-ah! Sto-op! Stop it pl-pleasuew...", "Go-o-o-od God-d-dpf!", "BZE-EW-EWQ! He-e-el-l-el!"))
+	addtimer(CALLBACK(src, .proc/malfunction_result), 2 SECONDS)
+
+
+//It's second proc, result of our malfunction
+/mob/living/simple_animal/hostile/hivemind/proc/malfunction_result()
+	if(prob(malfunction_chance))
+		apply_damage(rand(10, 25), BURN)
+
+
+//sometimes, players use closets, to staff mobs into it
+//and it's works pretty good, you just weld it and that's all
+//but not this time
+/mob/living/simple_animal/hostile/hivemind/proc/closet_interaction()
+	if(mob_size >= MOB_MEDIUM)
+		var/obj/structure/closet/closed_closet = loc
+		if(closed_closet && istype(closed_closet))
+			closed_closet.open(src)
+
+
+/mob/living/simple_animal/hostile/hivemind/say()
+	..()
+	playsound(src, pick('sound/machines/robots/robot_talk_heavy1.ogg',
+							'sound/machines/robots/robot_talk_heavy2.ogg',
+							'sound/machines/robots/robot_talk_heavy3.ogg',
+							'sound/machines/robots/robot_talk_heavy4.ogg',
+							'sound/machines/robots/robot_talk_light1.ogg',
+							'sound/machines/robots/robot_talk_light2.ogg',
+							'sound/machines/robots/robot_talk_light3.ogg',
+							'sound/machines/robots/robot_talk_light4.ogg',
+							'sound/machines/robots/robot_talk_light5.ogg',
+							), 50, 1)
+
+
+/mob/living/simple_animal/hostile/hivemind/Life()
+	//to prevent life ticking at dead mobs and repeatedly calling death(), an issue, i think
+	if(stat != CONSCIOUS)
```

Comment:
I believe life returns false when the mob is dead.

At the top of life code, you should have..

```
.=..()
if (.)
//The mob is alive, do stuff
```